docs/user/reference/hook.md

```diff
@@ -137,6 +142,236 @@ The result is that the events are consumed in the order: `2-1-3`. Beware.
+
+The following environment variables may be set depending on the exact hook:
+
+* $JUJU_RELATION holds the name of the relation.
+* $JUJU_RELATION_ID holds the ID of the relation.
+* $JUJU_REMOTE_UNIT holds the name of the related unit. 
+* $JUJU_REMOTE_APP holds the name of the related application.
+* $JUJU_DEPARTING_UNIT holds the name of a related unit departing the relation.
+
+### Secret hooks
+
+These hooks require an associated secret.
+The following environment variables may be set depending on the exact hook:
+
+* $JUJU_SECRET_ID holds the ID of the secret.
+* $JUJU_SECRET_LABEL holds the label of the secret.
+* $JUJU_SECRET_REVISION holds the revision of the secret.
+
+### Storage hooks
+
+These hooks require an associated storage.
+The hook names that these kinds represent will be prefixed by the storage name; for example, `database-storage-attached`.
+
+* $JUJU_STORAGE_ID holds the ID of the storage.
+* $JUJU_STORAGE_LOCATION holds the path at which the storage is mounted.
+* $JUJU_STORAGE_KIND holds the kind of storage (`block` or `filesystem`).
+
+### Upgrade series hook
+
+* $JUJU_TARGET_BASE holds the target base for the machine.
+
+### Workload (pebble) hooks
+
+These hooks require an associated workload/container.
+The hook names that these kinds represent will be prefixed by the workload/container name; for example, `mycontainer-pebble-ready`.
+
+The following environment variables may be set depending on the exact hook:
+ 
+* $JUJU_WORKLOAD_NAME holds the name of the container to which the hook pertains.
+* $JUJU_NOTICE_ID holds the Pebble notice ID.
+* $JUJU_NOTICE_TYPE holds the Pebble notice type.
+* $JUJU_NOTICE_KEY holds the Pebble notice key.
+* $JUJU_PEBBLE_CHECK_NAME holds the name of the Pebble check.
+
+## Hook ordering
+
+A charm's lifecycle consists of distinct **phases**:
+* install
+* operation
+* upgrade
+* tear down
+
+Generally, no assumptions can be made about order of hook execution. However, there are some limited guarantees
+about hook sequencing during install, upgrade and relation removal. 
+
+In normal operation, a unit will run at least the install, start, config-changed  and stop hooks over the course of its lifetime.
+
+### Install phase
+
+When a charm is first deployed, the following hook are executed in order before a charm reaches its operation phase:
+
+* storage-attached (if the unit has storage)
+* install
+* leader-elected (if the unit is a leader)
+* config-changed
+* start
+
+### Operation phase
+
+This phase is when a charm has completed its install operations and starts responding to events which correspond to interesting (relevant) changes to the Juju model.
+The behaviour is described in terms of hook kinds and is covered in subsequent sections.
+
+### Upgrade phase
+
+When a charm is upgraded, the `upgrade-charm` hook is usually followed by a `config-changed` hook.
```

Comment:
"usually" is vague. Is the uncertainty here whether it's _immediately_ followed, or something else?

Reply:
I can't recall any nuance here right now but I could be wrong. I'll remove "usually" until we decide it's needed.